### PR TITLE
[Unity][Disco] separate computation and communication into 2 stream

### PR DESCRIPTION
**Pull request by jinhongyii**
put computation on default stream and put communication on a new stream.

### CODE REVIEW

**Comment by jinhongyii** [comment]
cc: @junrushao

**Comment by junrushao** [comment]
i'm trying to figure out - which stream is used for compute and which is for communication? are we using null stream for compute?

**Comment by jinhongyii** [comment]
null stream (or default stream equivalently) is used for compute. The stream created in init_ccl is for communication

File: src/runtime/disco/nccl/nccl.cc [hunk 1]
```diff
@@ -91,26 +90,34 @@ void AllReduce(NDArray send, ReduceKind reduce_kind, NDArray recv) {
   NCCLThreadLocalContext* ctx = NCCLThreadLocalContext::Get();
   ShapeTuple shape = send.Shape();
   int64_t numel = shape->Product();
+  Device device = ctx->worker->default_device;
+  DeviceAPI::Get(device)->SyncStreamFromTo(device, nullptr, ctx->stream);
```
Comment (junrushao) [inline src/runtime/disco/nccl/nccl.cc:94]: one minor comment: please add a field `ctx->compute_stream` and rename `ctx->stream` to `ctx->comm_stream`

Comment (jinhongyii) [reply src/runtime/disco/nccl/nccl.cc:94]: makes sense

**Review by junrushao — APPROVED** [review]

**Pull request merged**